pgx/src/datum/array.rs

```diff
@@ -168,10 +195,23 @@ impl<'a, T: FromDatum> Array<'a, T> {
             .map(|nonnull| NullKind::Bits(unsafe { &*nonnull.as_ptr() }))
             .unwrap_or(NullKind::Strict(nelems));
 
+        // Assert correctness of our nullness checks and cleanup.
+        // SAFETY: Should be correctly constructed for slice validity.
+        let pallocd_null_slice = unsafe { slice::from_raw_parts(nulls, nelems) };
+        for i in 0..nelems {
+            assert_eq!(null_slice.get(i).unwrap(), pallocd_null_slice[i]);
+        }
+
+        // Throw away the slice we made.
+        mem::drop(pallocd_null_slice);
+        // SAFETY: We made it, we can break it. Or Postgres can, at least.
+        unsafe { pg_sys::pfree(nulls.cast()) };
+
```

Comment:
This code continues to call `deconstruct_array` to maintain compatibility for current schemes for now, but it immediately frees the pallocated slice of `bool` as unnecessary after checking that our `NullKind`-wrapped `BitSlice` "understands" the null mapping in the same way.